lib/sycamore/sycamore/reader.py

```diff
@@ -307,30 +307,37 @@ def opensearch(
                     f"Filter values must be a list of strings. Got {type(v)} instead. Please provide a list of values."
                 )
 
-        filter = None
-
         # Allow the OpenSearchReader to use the result filter if it does not conflict
         # with an existing filter present in the passed-in query.
         if result_filter is not None:
             if "knn" in query["query"]:
-                if "filter" not in query["query"]["knn"]:
-                    filter = result_filter
-            elif not ("bool" in query["query"] and "must" in query["query"]["bool"]):
-                filter = result_filter
+                if "filter" in query["query"]["knn"]:
+                    raise ValueError("'query' cannot contain a filter when 'result_filter' is provided.")
```

Comment:
why isn't this:
```
else:
  raise ValueError("Unable to figure out how to add result filter to query: {query['query']}")
```
That makes sure that if we can't apply the filter in a way we expect it to work we will stop.